src/main/java/seedu/address/model/gameentry/DatePlayed.java

```diff
@@ -22,33 +27,61 @@ public DatePlayed() {
     }
 
     /**
-     * Constructs DatePlayed object.
+     * Constructs DatePlayed object from a String.
      *
-     * @param date Date that this object represents.
-     * @param isTimeIndicated boolean to indicate whether to ignore minute timestamp.
+     * @param datePlayedString String representing DatePlayed.
      */
-    public DatePlayed(Date date, boolean isTimeIndicated) {
-        this.datePlayed = date;
-        this.isTimeIndicated = isTimeIndicated;
-    }
+    public DatePlayed(String datePlayedString) throws InvalidDateFormatException {
+        requireNonNull(datePlayedString);
+        if (!isValidDatePlayedString(datePlayedString)) {
```

Comment:
Refer to [Address](https://github.com/nus-cs2103-AY2122S1/tp/blob/master/src/main/java/seedu/address/model/person/Address.java) in AB3.

Format checking is done by [AppUtil::checkArguments](https://github.com/nus-cs2103-AY2122S1/tp/blob/master/src/main/java/seedu/address/commons/util/AppUtil.java) and it throws the RuntimeException there.

Reply:
> Okay so this is throwing InvalidDateFormatException. What's the reason to have a subclass of IllegalArgumentException hold the error message instead of having DatePlayed pass the error message to AppUtil?

Yep I think you are right. I will have this removed in my next PR.